visualization/app/codeCharta/state/store/lookUp/pathToNode/pathToNode.service.ts

```diff
@@ -0,0 +1,24 @@
+import { StoreService } from "../../../store.service"
+import { IRootScopeService } from "angular"
+import { setPathToNode } from "./pathToNode.actions"
+import { CodeMapNode } from "../../../../codeCharta.model"
+import { CodeMapPreRenderService, CodeMapPreRenderServiceSubscriber } from "../../../../ui/codeMap/codeMap.preRender.service"
+import { hierarchy } from "d3"
+
+export class PathToNodeService implements CodeMapPreRenderServiceSubscriber {
+	constructor(private $rootScope: IRootScopeService, private storeService: StoreService) {
+		CodeMapPreRenderService.subscribe(this.$rootScope, this)
+	}
+
+	public onRenderMapChanged(map: CodeMapNode) {
+		const pathToNode = new Map<string, CodeMapNode>()
+		pathToNode.set(map.path, map)
+		hierarchy(map)
+			.descendants()
+			.forEach(x => {
+				pathToNode.set(x.data.path, x.data)
```

Comment:
Ar you aware of that a nodes identity is not only defined by the path but by the path and the type? There can be a file and a folder of the same name.

Reply:
Is that really the case? This would require a folder name and a file name to be the same while the file has no extension at all. With the file extension we should have unique file and folder names, don't we?

Reply:
Yea. Well a file does not need an extension. And we can not assume that codecharta is only used for programming languages. Codecharta was from the beginning a generic tool, which can handle every valid cc.json file

Reply:
I remember the idea of the generic approach, but I don't think that it will ever be used outside of coding context. I mean, it's even called CODEcharta and we have a treeView for folders and files.

Reply:
That might be. Still, a file even if it contains code does not always need a file extension. The treeView is super generic?! Folders are not only used for programming?! 😁

Truely: it might happen super seldom, but still we already have functions in our codebase to get a specific node and, until now, we always use the path and type.. 

I vote for consistency, no matter which option we choose. Though, not accepting files with identical paths were a breaking change in file validation

Reply:
This is resolved now that we use ids. Great suggestion!